### PR TITLE
fix: weaviate da del behavior

### PR DESCRIPTION
**Pull request by winstonww**
Align the del/destructor behavior of `DocumentArrayWeaviate` with `DocumentArraySqlite` (delete data only when `self._persist` is `True`, that is when table name/name in provided in `config`).

Also, adding a check to assert `name` provided is capitalized. This is because if `name` is not capitalized, Weaviate will automatically capitalize the name but users may not be aware of this special case.

### CODE REVIEW

**Review by JoanFM — CHANGES_REQUESTED** [review]
What is this del doing? Are we sure this behavior is desired when garbage collector kicks in?

**Comment by winstonww** [comment]
@JoanFM Yes, an issue is indeed observed and is under investigation detailed in the discussion [here](https://jinaai.slack.com/archives/C02AC4T8Y5T/p1644324667978449). Thoughts appreciated, thanks

File: docarray/array/storage/weaviate/backend.py [hunk 1]
```diff
@@ -65,6 +65,11 @@ def _init_storage(
         self._n_dim = config.n_dim
         self._serialize_config = config.serialize_config
 
+        if config.name and config.name != config.name.capitalize():
+            raise ValueError('weaviate class name has to be capitalized')
```
Comment (numb3r3) [inline docarray/array/storage/weaviate/backend.py:69]: Given this error report, the user would still not know how to correct the configurations. Could you give more clues in the output?

Comment (winstonww) [reply docarray/array/storage/weaviate/backend.py]: Yep, addressed thank you.

File: docarray/array/storage/registry.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+from collections import defaultdict
+
+
+_REGISTRY = defaultdict(lambda: defaultdict(list))
```
[the remark this answers is not in the document]
Comment (winstonww) [reply docarray/array/storage/registry.py:4]: This is suggested by @alaeddine-13  to fix the issue detailed [here](https://jinaai.slack.com/archives/C02AC4T8Y5T/p1644399473166809?thread_ts=1644324667.978449&cid=C02AC4T8Y5T)/

**Review by alaeddine-13 — APPROVED** [review]
LGTM

**Pull request merged**